crates/librqbit/src/blocklist.rs

```diff
@@ -0,0 +1,314 @@
+            trace!("Detected Gzip file, decompressing...");
+            Box::pin(BufReader::new(GzipDecoder::new(reader)))
+        } else {
+            trace!("Plain text file detected.");
+            Box::pin(reader)
+        };
+
+        let reader = BufReader::new(reader);
+        let mut lines = reader.lines();
+        let mut ip_ranges: Vec<std::ops::Range<IpAddr>> = Vec::new();
+        while let Some(line) = lines.next_line().await? {
+            // Skip comments and empty lines
+            if line.starts_with('#') || line.trim().is_empty() {
+                continue;
+            }
+
+            if let Some((start_ip, end_ip)) = parse_ip_range(&line) {
+                let range = start_ip..(increment_ip(end_ip).unwrap());
+                ip_ranges.push(range);
+            }
+        }
+
+        info!(
+            ip_entry_count = ip_ranges.len(),
+            "Finished loading blocklist"
+        );
+
+        let blocklist = Self::new(ip_ranges);
+        Ok(blocklist)
+    }
+
+    pub fn is_blocked(&self, ip: &IpAddr) -> bool {
+        self.blocked_ranges.query_point(*ip).next().is_some()
+    }
+}
+
+/// Safely increments an `IpAddr`, returning `None` if it would overflow.
+fn increment_ip(ip: IpAddr) -> Option<IpAddr> {
+    match ip {
+        IpAddr::V4(ipv4) => {
+            let num = u32::from_be_bytes(ipv4.octets());
+            num.checked_add(1).map(|n| IpAddr::V4(Ipv4Addr::from(n)))
+        }
+        IpAddr::V6(ipv6) => {
+            let num = u128::from_be_bytes(ipv6.octets());
+            num.checked_add(1).map(|n| IpAddr::V6(Ipv6Addr::from(n)))
+        }
+    }
+}
+
+fn parse_ip_range(line: &str) -> Option<(IpAddr, IpAddr)> {
+    // Skip comments and empty lines
+    let line = line.trim();
+    if line.starts_with('#') || line.is_empty() {
+        return None;
+    }
+
+    let is_ipv4 = line.matches('.').count() >= 6;
+    // Find the split point based on whether it's IPv4 or not
+    let split_point: usize = if is_ipv4 {
+        line.rfind(':')
+    } else {
+        line.find(':')
+    }
+    .unwrap_or(0);
+
+    let (rule_name, ip_range) = line.split_at(split_point + 1);
+    if let Some((start, end)) = ip_range.split_once('-') {
+        if let (Ok(start_ip), Ok(end_ip)) =
+            (IpAddr::from_str(start.trim()), IpAddr::from_str(end.trim()))
+        {
+            return Some((start_ip, end_ip));
+        } else {
+            // Mismatched IP versions, skip this range
+            debug!(rulen_name = rule_name, "Could not be parsed");
+        }
+    }
+
+    None
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use async_compression::tokio::write::GzipEncoder;
+    use mockito::{Server, ServerGuard};
+    use std::thread::{self, JoinHandle};
+    use tokio::io::AsyncWriteExt;
+
+    struct TestServer {
+        server: ServerGuard,
+        mock: mockito::Mock,
+        url: String,
+        _thread: JoinHandle<()>,
+    }
+
+    impl TestServer {
```

Comment:
How about removing HTTP servers and just test parsing streams?
Just to decrease complexity of this test a bit and remove mockito which wasn't necessary so far?